GenderPayGap.WebUI/BackgroundJobs/ScheduledJobs/AnonymiseThreeYearOldFeedbackJob.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using GenderPayGap.Core.Interfaces;
+using GenderPayGap.Database.Models;
+using GenderPayGap.Extensions;
+
+namespace GenderPayGap.WebUI.BackgroundJobs.ScheduledJobs
+{
+    public class AnonymiseThreeYearOldFeedbackJob
+    {
+
+        private readonly IDataRepository dataRepository;
+
+        public AnonymiseThreeYearOldFeedbackJob(IDataRepository dataRepository)
+        {
+            this.dataRepository = dataRepository;
+        }
+
+        public void AnonymiseFeedbackAction()
+        {
+            var runId = JobHelpers.CreateRunId();
+            var startTime = VirtualDateTime.Now;
+            JobHelpers.LogFunctionStart(runId, nameof(AnonymiseFeedbackAction), startTime);
+
+            try
+            {
+                DateTime threeYearsAgo = VirtualDateTime.Now.AddYears(-3);
+
+                List<Feedback> feedback = dataRepository.GetAll<Feedback>()
+                    .Where(f => DateTime.Compare(f.CreatedDate, threeYearsAgo) <= 0)
+                    .Where(f => !f.HasBeenAnonymised)
+                    .ToList();
+
+                foreach (Feedback feedbackItem in feedback)
+                {
+                    AnonymiseFeedback(feedbackItem);
+                }
+
+                JobHelpers.LogFunctionEnd(runId, nameof(AnonymiseFeedbackAction), startTime);
+            }
+            catch (Exception ex)
+            {
+                JobHelpers.LogFunctionError(runId, nameof(AnonymiseFeedbackAction), startTime, ex);
+
+                //Rethrow the error
+                throw;
+            }
+        }
+
+        public void AnonymiseFeedback(Feedback feedback)
+        {
+            feedback.OtherSourceText = string.IsNullOrWhiteSpace(feedback.OtherSourceText)
+                ? "not supplied"
+                : "supplied";
+
+            feedback.OtherReasonText = string.IsNullOrWhiteSpace(feedback.OtherReasonText)
+                ? "not supplied"
+                : "supplied";
+
+            feedback.OtherPersonText = string.IsNullOrWhiteSpace(feedback.OtherPersonText)
+                ? "not supplied"
+                : "supplied";
+
+            feedback.EmailAddress = string.IsNullOrWhiteSpace(feedback.EmailAddress)
+                ? "not supplied"
+                : "supplied";
+
+            feedback.PhoneNumber = string.IsNullOrWhiteSpace(feedback.PhoneNumber)
+                ? "not supplied"
+                : "supplied";
+
+            feedback.Details = string.IsNullOrWhiteSpace(feedback.Details)
+                ? "not supplied"
+                : "supplied";

```

Comment:
I wanted to refactor this into something a bit cleaner, but I didn't want to add unnecessary complexity to something that's not too long and actually pretty readable! I'm not sure if there's a nicer solution than this - I'd love to be able to use JavaScript style `feedback["property name"]` syntax or something similar but [this post](https://stackoverflow.com/questions/10283206/setting-getting-the-class-properties-by-string-name) suggests that it might cause a bunch of overhead that's maybe not worth it to simplify 22 lines of code?

Reply:
We could do something like

```csharp
...
feedback.OtherSourceText = Anonymise(feedback.OtherSourceText);
feedback.Details = Anonymise(feedback.Details);
...

private static string Anonymise(string original) {
    return string.IsNullOrWhitespace(original)
        ? "not supplied"
        : "supplied";
}
```